### PR TITLE
LMSVN stuff for orchestration

### PR DESCRIPTION
**Pull request by rushtong**
See  https://broadinstitute.atlassian.net/browse/GAWB-1352
Companion UI PR: https://github.com/broadinstitute/firecloud-ui/pull/600

Add a new property that is not meant to be sent to ES nor is meant to be user-editable. 

---
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [x] **Product Owner** sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Squash commits and merge to develop
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [x] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [x] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [x] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/ElasticSearchDAOSupport.scala [hunk 1]
```diff
@@ -44,18 +44,21 @@ trait ElasticSearchDAOSupport extends LazyLogging {
...
     }
-    ESDatasetProperty(maps).toJson.prettyPrint
+    ESDatasetProperty(attributeDetailMap).toJson.prettyPrint
   }
 
-  def detailFromAttribute(label: String, detail: AttributeDetail) = {
+  def detailFromAttribute(label: String, detail: AttributeDetail): (String, ESPropertyFields) = {
+    val itemType = detail match {
+      case x if x.`type` == "array" && x.items.isDefined => x.items.get.`type`
+      case x => x.`type`
```
Comment (rushtong) [inline src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/ElasticSearchDAOSupport.scala]: this needs to be safer, something like `case _ => detail.type`

File: src/main/scala/org/broadinstitute/dsde/firecloud/model/ModelJsonProtocol.scala [hunk 2]
```diff
@@ -5,7 +5,7 @@ import spray.http.StatusCode
 import spray.http.StatusCodes.BadRequest
 import org.broadinstitute.dsde.firecloud.model.MethodRepository.{AgoraPermission, FireCloudPermission}
 import spray.json._
-import spray.json.DefaultJsonProtocol._
+import spray.json.DefaultJsonProtocol.{jsonFormat4, _}
```
Comment (rushtong) [inline src/main/scala/org/broadinstitute/dsde/firecloud/model/ModelJsonProtocol.scala]: Unnecessary

File: src/main/resources/library/attribute-definitions.json [hunk 3]
```diff
@@ -257,7 +265,8 @@
         {"property": "library:broadInternalResearchProjectID"},
         {"property": "library:broadInternalResearchProjectName"},
         {"property": "library:broadInternalCohortName"},
-        {"property": "library:broadInternalSeqProjectNumbers", "inputHint": "Broad internal IDs"}
+        {"property": "library:broadInternalSeqProjectNumbers", "inputHint": "Broad internal IDs"},
+        {"property": "library:lmsvn", "editable": false}
```
Comment (skwalker) [inline src/main/resources/library/attribute-definitions.json:268]: does editable need to be here and above (line 211)? I feel like it probably only needs to be in one place?

File: src/main/resources/library/attribute-definitions.json [hunk 4]
```diff
@@ -201,6 +201,14 @@
     },
     "library:orsp": {
       "type": "string"
+    },
+    "library:lmsvn": {
+      "title": "Library Metadata Schema Version Number",
+      "description": "Library Metadata Schema Version Number",
+      "type": "integer",
+      "default": 1,
```
Comment (skwalker) [inline src/main/resources/library/attribute-definitions.json:209]: Should we be starting at 0? (just looking at 1417)

File: src/main/scala/org/broadinstitute/dsde/firecloud/model/ElasticSearch.scala [hunk 5]
```diff
@@ -9,7 +9,12 @@ import org.broadinstitute.dsde.firecloud.model.ModelJsonProtocol._
 
 case class AttributeDefinition(properties: Map[String, AttributeDetail])
 
-case class AttributeDetail(`type`: String, items: Option[AttributeDetail] = None, aggregate: Option[Boolean] = None)
+case class AttributeDetail(
+  `type`: String,
+  items: Option[AttributeDetail] = None,
+  aggregate: Option[Boolean] = None,
+  indexable: Option[Boolean] = Some(true)
```
Comment (davidangb) [inline src/main/scala/org/broadinstitute/dsde/firecloud/model/ElasticSearch.scala]: I'd prefer that `aggregate` and `indexable` be consistent: either both default to None, or both default to Some.

**Review by davidangb — COMMENTED** [review]
one minor comment inline, otherwise 👍

File: src/main/scala/org/broadinstitute/dsde/firecloud/model/ElasticSearch.scala (hunk 5 above)
Comment (rushtong) [reply src/main/scala/org/broadinstitute/dsde/firecloud/model/ElasticSearch.scala]: 👍

**Review by skwalker — APPROVED** [review]
works well, 2 tiny inline comments

File: src/main/resources/library/attribute-definitions.json (hunk 3 above)
Comment (rushtong) [reply src/main/resources/library/attribute-definitions.json:268]: Good catch ... it was originally only in one place, but with Matt's refactor, I needed to add it here and didn't remove it from above. I'm ambivalent about having these kinds of property attributes spread out at all - I'd really like these to be one thing so we don't have duplication.

File: src/main/resources/library/attribute-definitions.json (hunk 4 above)
Comment (rushtong) [reply src/main/resources/library/attribute-definitions.json:209]: GAWB-1417 describes going from v1 to v2, so I think this being v1 is correct.

Comment (skwalker) [reply src/main/resources/library/attribute-definitions.json:209]: oh oops I thought I saw 0 someone but never mind!

File: src/main/resources/library/attribute-definitions.json (hunk 3 above)
Comment (skwalker) [reply src/main/resources/library/attribute-definitions.json:268]: yeah, I don't really like the duplication that's there now either, but thanks for this fix! PR looks good to me 👍

Comment (davidangb) [reply src/main/resources/library/attribute-definitions.json:268]: could probably add this in to GAWB-1389.

**Review by davidangb — APPROVED** [review]
👍 based on latest changes!

File: src/main/resources/library/attribute-definitions.json (hunk 3 above)
Comment (rushtong) [reply src/main/resources/library/attribute-definitions.json:268]: Added to GAWB-1389, thanks David!

**Pull request merged**